go/vt/tabletmanager/state_change.go

```diff
@@ -293,6 +293,27 @@ func (agent *ActionAgent) changeCallback(ctx context.Context, oldTablet, newTabl
 		}
 	}
 
+	// See if we need to enable or disable semi-sync replication.
+	// Note we never enable the master side, as that would have
+	// been done by either the reparent process, or an external
+	// tool (like Orchestrator).
+	if *enableSemiSync {
+		wasMasterEligible := isMasterEligible(oldTablet.Type)
+		isMasterEligible := isMasterEligible(newTablet.Type)
+		if wasMasterEligible && !isMasterEligible {
+			if err := agent.MysqlDaemon.SetSemiSyncEnabled(false, false); err != nil {
+				log.Errorf("Failed to SetSemiSyncEnabled(false, false), may not ack correctly: %v", err)
+			}
+		}
+		if !wasMasterEligible && isMasterEligible && newTablet.Type != topodatapb.TabletType_MASTER {
+			// We go from any type to REPLICA, let's
+			// enable semi-sync acks.
+			if err := agent.MysqlDaemon.SetSemiSyncEnabled(false, true); err != nil {
```

Comment:
Note that MySQL won't actually start sending semi-sync acks here unless you stop slave and then start it again. And I believe #2195 is exactly about that, i.e. rpl_semi_sync_slave_enabled was somehow set already, but there was no StopSlave+StartSlave for it to start working.